### PR TITLE
feat(Avatar): added square prop

### PR DESCRIPTION
**Pull request by mnajdova**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

(give an overview)

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/12277)

### CODE REVIEW

File: packages/fluentui/react/src/themes/teams/components/Avatar/avatarStyles.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { ComponentSlotStylesPrepared, ICSSInJSStyle } from '@fluentui/styles';
 import { AvatarProps } from '../../../../components/Avatar/Avatar';
 import { AvatarVariables } from './avatarVariables';
 
-export type AvatarStylesProps = Pick<AvatarProps, 'size'>;
+export type AvatarStylesProps = Pick<AvatarProps, 'size' | 'square'>;
```
Comment (layershifter) [inline packages/fluentui/react/src/themes/teams/components/Avatar/avatarStyles.ts]: Let's move this to the component as we are here 🙂

File: packages/fluentui/react/src/themes/teams/components/Avatar/avatarStyles.ts [hunk 2]
```diff
@@ -28,27 +28,34 @@ const avatarStyles: ComponentSlotStylesPrepared<AvatarStylesProps, AvatarVariabl
...
       height: sizeInRem,
       lineHeight: sizeInRem,
-      fontSize: pxToRem(sizeToPxValue[size] / 2.333),
+      fontSize: pxToRem(sizeToPxValue[p.size] / 2.333),
       verticalAlign: 'top',
       textAlign: 'center',
-      padding: '0px'
+      padding: '0px',
+      ...(p.square && {
+        borderRadius: v.avatarBorderRadius
```
Comment (layershifter) [inline packages/fluentui/react/src/themes/teams/components/Avatar/avatarStyles.ts]:
```suggestion
        borderRadius: v.squareBorderRadius
```

Should not contain prop's name?

File: packages/fluentui/react/src/themes/teams/components/Avatar/avatarVariables.ts [hunk 3]
```diff
@@ -1,13 +1,17 @@
...
   avatarBorderWidth: string;
+  squareAvatarBorderRadius: string;
   statusBorderColor: string;
   statusBorderWidth: string;
 }
 
 export default (siteVariables): AvatarVariables => ({
   avatarBorderColor: '',
   avatarBorderWidth: '0',
+  squareAvatarBorderRadius: pxToRem(4),
```
Comment (layershifter) [inline packages/fluentui/react/src/themes/teams/components/Avatar/avatarVariables.ts]: Just wondering why `squareAvatarBorderRadius`? As I remember convention is `${PROP}${SLOT}${RULE}` 🤔

Comment (mnajdova) [reply packages/fluentui/react/src/themes/teams/components/Avatar/avatarVariables.ts]: This component is awkward as there is a root, and than avatar and status part. The avatar part can be image, label, soon also icon, I guess that's why it was chosen to be like this... I am just following previous convention, didn't want to change all variables names..

**Pull request merged**